backend/src/api/automation/automationCreate.ts

```diff
@@ -19,9 +24,25 @@ import track from '../../segment/track'
  */
 export default async (req, res) => {
   new PermissionChecker(req).validateHas(Permissions.values.automationCreate)
+
+  if (!(await isFeatureEnabled(FeatureFlag.AUTOMATIONS, req.currentTenant.id, req.posthog))) {
+    await req.responseHandler.error(
+      req,
+      res,
+      new Error403(req.language, 'entities.automation.errors.planLimitExceeded'),
+    )
+    return
+  }
+
   const payload = await new AutomationService(req).create(req.body.data)
 
   track('Automation Created', { ...payload }, { ...req })
 
+  identifyTenant(req)
+
+  // wait a small window for posthog
+  // to process the queue message before returing back
+  await timeout(100)
```

Comment:
Is this absolutely necessary? I really don't like having random waiting because something else needs processing :/.